src/scheduler.cpp

```diff
@@ -23,7 +23,8 @@ CScheduler::~CScheduler()
 #if BOOST_VERSION < 105000
 static boost::system_time toPosixTime(const boost::chrono::system_clock::time_point& t)
 {
-    return boost::posix_time::from_time_t(boost::chrono::system_clock::to_time_t(t));
+    auto millis = boost::chrono::duration_cast<boost::chrono::milliseconds>(t.time_since_epoch()).count() % 1000;
+    return boost::posix_time::from_time_t(boost::chrono::system_clock::to_time_t(t)) + boost::posix_time::milliseconds(millis);
```

Comment:
```from_time_t()``` (which seems to be the only reasonable way we have to create the posix_time without involving clocks, strings, or locales) is seconds since the epoch, so we lose the milliseconds there. The change here grabs them off of the input time, and adds them back after creating the posix_time.

> What is the reason that this won't work directly?
> ```
> return boost::posix_time::milliseconds(boost::chrono::duration_cast<boost::chrono::milliseconds>(t.time_since_epoch()).count());
> ```

This would return a duration rather than a time point. It would work if it were added to a time point at the epoch, though:
```
return boost::posix_time::from_time_t(0) + boost::posix_time::milliseconds(boost::chrono::duration_cast<boost::chrono::milliseconds>(t.time_since_epoch()).count());
```
Tested both, either is fine by me. Do you have a preference?

I'll add a comment either way.